doc/user/compile.md

```diff
@@ -55,6 +55,7 @@ program (before including mlpack or Armadillo!).
 | `-DMLPACK_CERR_STREAM=std::cerr` | `#define MLPACK_CERR_STREAM std::cerr` | Set the default error stream.  (Defaults to `std::cerr`.) |
 | `-DMLPACK_PRINT_INFO` | `#define MLPACK_PRINT_INFO` | Print information messages (`[INFO ]`) during program execution. |
 | `-DMLPACK_PRINT_WARN` | `#define MLPACK_PRINT_WARN` | Print warning messages (`[WARN ]`) during program execution. |
+| `-DMLPACK_NO_STD_MUTEX` | `define MLPACK_NO_STD_MUTX` | Disable mutex inside mlpack if you have one core, or no support for mutex. |
```

Comment:
```suggestion
| `-DMLPACK_NO_STD_MUTEX` | `#define MLPACK_NO_STD_MUTEX` | Disable mutexes inside mlpack; use this if your system has no support for `std::mutex` and has only one core.  You may also need to define `ARMA_DO_NOT_USE_STD_MUTEX` for Armadillo. |
```

I added the bit about Armadillo because if a user includes Armadillo first and then mlpack, then defining `ARMA_DO_NOT_USE_STD_MUTEX` in `mlpack/config.hpp` won't do anything (because Armadillo was already included).

I also changed the language just a little bit, I don't want to imply that all users with one core should always disable mutexes.

Also, do you think you could move this to the `Functionality` section, maybe under `MLPACK_DISABLE_STB`?  It doesn't really fit in the `Output` section.